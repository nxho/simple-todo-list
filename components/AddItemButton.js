import React from 'react';
import {
	StyleSheet,
	Text,
	TouchableOpacity,
	View,
} from 'react-native';

export default function AddItemButton(props) {
	return(
		<TouchableOpacity style={styles.container} onPress={props.onPress}>
			<Text style={styles.text}>+</Text>
		</TouchableOpacity>
	);
}

const styles = StyleSheet.create({
	container: {
		height: 65,
		width: 65,
		borderRadius: 100,
		alignSelf: 'center',
		alignItems: 'center',
		justifyContent: 'center',
		backgroundColor: '#50E3A4',
	},
	text: {
		textAlign: 'center',
		textAlignVertical: 'center',
		lineHeight: 35,
		fontSize: 35,
		color: '#46BD8A'
	},
});

