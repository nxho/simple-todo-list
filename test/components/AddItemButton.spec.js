import React from 'react';
import { shallow } from 'react-native-testing-library';
import AddItemButton from '../../components/AddItemButton';

describe('AddItemButton component', () => {
	it('renders correctly', () => {
		const { output } = shallow(<AddItemButton />);

		expect(output).toMatchSnapshot();
	});
});
