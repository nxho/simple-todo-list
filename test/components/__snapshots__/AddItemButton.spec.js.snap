// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItemButton component renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#50E3A4",
      "borderRadius": 100,
      "height": 65,
      "justifyContent": "center",
      "width": 65,
    }
  }
>
  <Component
    style={
      Object {
        "color": "#46BD8A",
        "fontSize": 35,
        "lineHeight": 35,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    +
  </Component>
</TouchableOpacity>
`;
